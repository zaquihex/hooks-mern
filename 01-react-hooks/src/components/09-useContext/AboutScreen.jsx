import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const AboutScreen = () => {
    const { user,setUser } = useContext(UserContext);
    return (
        <div>
            <h1>AboutScreen</h1>
            <hr/>
            <pre>{JSON.stringify(user, null, 3)}</pre>
            <button 
                className="container btn btn-warning"
                onClick={()=> setUser(null)}
            >
                Logout
            </button>
        </div>
    )
}
