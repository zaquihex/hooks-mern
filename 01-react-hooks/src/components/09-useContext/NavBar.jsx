import React from 'react'
import { Link, NavLink, Redirect } from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <NavLink exact activeClassName="active" exact to="/home" className="navbar-brand" href="#">useContext</NavLink>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink activeClassName="active" exact to="/home" className="nav-item nav-link" href="#">Home</NavLink>
                    <NavLink activeClassName="active" exact to="/about" className="nav-item nav-link" href="#">About</NavLink>
                    <NavLink activeClassName="active" exact to="/login" className="nav-item nav-link" href="#">Login</NavLink>
                    <Redirect to="/home" />
                </div>
            </div>
        </nav>
    )
}
