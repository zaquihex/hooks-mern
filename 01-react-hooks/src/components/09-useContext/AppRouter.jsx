import React from 'react';
import {NavBar} from "./NavBar";
import {HomeScreen} from "./HomeScreen";
import {AboutScreen} from "./AboutScreen";
import {LoginScreen} from "./LoginScreen";

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Switch>
                    <Route exact path="/home" component={HomeScreen} />
                    <Route exact path="/about" component={AboutScreen} />
                    <Route exact path="/login" component={LoginScreen} />
                    <Redirect path="/home" />
                </Switch>

            </div>
        </Router>
    )
}