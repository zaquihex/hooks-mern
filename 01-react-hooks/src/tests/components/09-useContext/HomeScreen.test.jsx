import React, {useContext} from 'react';
import {shallow, mount} from 'enzyme';
import {HomeScreen} from "../../../components/09-useContext/HomeScreen";
import {UserContext} from "../../../components/09-useContext/UserContext";


describe('Pruebas en <HomeScreen />', ()=> {
    const user = {
        user: 'zaquiel',
        email: 'zaquihex@gmail.com'
    };
    const wrapper = mount(
        <UserContext.Provider value={{user}}>
        <HomeScreen />
        </UserContext.Provider>);

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

});