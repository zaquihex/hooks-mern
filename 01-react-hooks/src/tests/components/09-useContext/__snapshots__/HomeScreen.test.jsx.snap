// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
   "user": "zaquiel",
   "email": "zaquihex@gmail.com"
}
  </pre>
</div>
`;
